openwisp_firmware_upgrader/tests/test_private_storage.py

```diff
@@ -74,3 +80,44 @@ def test_superuser(self):
         user = self._get_admin()
         self.client.force_login(user)
         self._download_firmware_assert_status(200)
+
+    def test_view_permission_check(self):
+        staff_user = self._get_operator()
+        self.client.force_login(staff_user)
+        org = self.image.build.category.organization
+        self._download_firmware_assert_status(403)
+
+        # Add view permission first
+        content_type = ContentType.objects.get_for_model(FirmwareImage)
+        perm_codename = get_permission_codename('view', FirmwareImage._meta)
+        view_perm = Permission.objects.get(
+            content_type=content_type, codename=perm_codename
+        )
+        staff_user.user_permissions.add(view_perm)
+
+        self._create_org_user(user=staff_user, organization=org, is_admin=True)
+        self._download_firmware_assert_status(200)
+
+        # Remove org manager status
+        org_user = staff_user.openwisp_users_organization.get(
+            organization_users__organization=org
+        )
+        org_user.is_admin = False
+        org_user.save()
+
+        # Remove staff status
+        staff_user.is_staff = False
+        staff_user.save()
+
+        # Restore org manager status
+        org_user.is_admin = True
+        org_user.save()
+
+        # No access (missing staff status)
+        self._download_firmware_assert_status(403)
+
+        # Remove view permission
+        staff_user.user_permissions.remove(view_perm)
+
+        # No access (missing view permission)
+        self._download_firmware_assert_status(403)
```

Comment:
Looks good, can you refactor the code to use `subTest`? It helps to organize code for testing different scenarios. You can look around the code for reference. 

https://github.com/search?q=repo%3Aopenwisp%2Fopenwisp-firmware-upgrader+subTest&type=code